synapse/visibility.py

```diff
@@ -643,14 +643,13 @@ def check_event_is_visible(
     # this check but would base the filtering on an outdated view of the membership events.
 
     partial_state_invisible_events = set()
-    if filter_out_erased_senders:
```

Comment:
Am I missing something here?

Reply:
I have put some [comments](https://github.com/matrix-org/synapse/commit/1526ff389f02d14d0df729bd6ea35836e758c449#r103679644) regarding that in the original PR.
We perhaps want to use a dedicated boolean like `filter_partial_state` for this (and keep your rename to `filter_out_erased_senders` for the rest of the code).
I think the logic I had in mind is that we don't care about having non local events when checking if we should use an event as an extremity for backfill: the remote server would then filter on his side anyway what need to be filtered, with a (potentially) a more current view than the joining server.